src/hooks/useTableOrder.ts

```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react'
+import {
+    getNextSortingParameters,
+    ScrollTableOrderDirection,
+} from '~/shared/components/ScrollTable/ScrollTable'
+
+export const useTableOrder = () => {
+    const [orderBy, setOrderBy] = useState<string | undefined>()
+    const [orderDirection, setOrderDirection] = useState<
+        ScrollTableOrderDirection | undefined
+    >()
+    const handleOrderChange = (columnKey: string) => {
+        const newOrderSettings = getNextSortingParameters(
+            orderBy,
+            columnKey,
+            orderDirection,
+        )
+        setOrderBy(newOrderSettings.orderBy)
+        setOrderDirection(newOrderSettings.orderDirection)
+    }
+
+    return {
+        orderBy,
+        orderDirection,
+        handleOrderChange,
+    }
+}
```

Comment:
Let's tweak it so that
- a ref to `handleOrderChange` stays the same between re-renders, and
- we do less for more.

Like so, for example:

```suggestion
import { useCallback, useState } from 'react'
import { getNextSortingParameters } from '~/shared/components/ScrollTable/ScrollTable'

export const useTableOrder = () => {
    const [order, setOrder] = useState<ReturnType<typeof getNextSortingParameters>>({
        orderBy: undefined,
        orderDirection: undefined,
    })

    const handleOrderChange = useCallback((columnKey: string) => {
        setOrder(({ orderBy, orderDirection }) =>
            getNextSortingParameters(orderBy, columnKey, orderDirection),
        )
    }, [])

    return {
        ...order,
        handleOrderChange,
    }
}
```